agent/utils/utils.go

```diff
@@ -105,6 +106,31 @@ type ForwardedRequest struct {
 	Contents *http.Request
 }
 
+// CookieCache represents a LRU cache to store session ID -> CookieJar
+type CookieCache struct {
+	cache *lru.Cache
+	sync.Mutex
```

Comment:
When wrapping a sync.Mutex, it is best to have the type not be exported outside of the package. The reason is that prevents outside code from erroneously locking or unlocking the mutex.

In practice, one way you can do that is by embedding the mutex as a field instead of extending the struct.

e.g.

```go
type CookieCache struct {
	cache *lru.Cache
	mu sync.Mutex
}
```

... then, instead of writing `cj.Lock()` or `cj.Unlock()`, you would write `cj.mu.Lock()` or `cj.mu.Unlock()`